File: code_blocks/integrations/third-party-integrations/mparticle_1.dart
import 'package:mparticle_flutter/mparticle_flutter.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> setup() async {
  await Purchases.configure(
    PurchasesConfiguration('public_sdk_key')..appUserId = 'my_app_user_id',
  );
  final options = MParticleOptions(
    apiKey: 'API_KEY',
    apiSecret: 'API_SECRET',
    identityRequest: IdentityRequest(userId: 'my_app_user_id'),
  );
  await MParticleFlutter.start(options);
}

File: code_blocks/integrations/third-party-integrations/mparticle_1.kt
import com.mparticle.MParticle
import com.mparticle.MParticleOptions
import com.mparticle.identity.IdentityApiRequest
import com.revenuecat.purchases.Purchases

val identityRequest = IdentityApiRequest.withEmptyUser().also {
    it.userId = "my_app_user_id"
}

val options = MParticleOptions.builder(this)
    .credentials("API_KEY", "API_SECRET")
    .identity(identityRequest)
    .build()

MParticle.start(this, options)
Purchases.configure(this, "public_sdk_key", appUserID = "my_app_user_id")

